executor/src/witgen/mod.rs

```diff
@@ -82,11 +83,26 @@ where
         machines,
         query_callback,
     };
+
+    // If the main machine is a VM that has been compiled from Powdr ASM, there
+    // will be a `main.instr_return` witness column that represents the latch.
```

Comment:
Do we actually need a latch if this is the main machine?

Otherwise, if it is not the main machine, then we have a selector in the lookup we can use for latch, don't we?

Reply:
Yes, but at this point it is always the main machine (secondary VMs are instantiated in `executor/src/witgen/machines/machine_extractor.rs`).

Basically, if there is a latch, we'll be able to set a default operation ID after the main return. Otherwise, the constraints have to determine a unique witness.